src/App.tsx

```diff
@@ -1,10 +1,25 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    {/* Keep this input for tests */}
-    <input type="text" />
+export const App: React.FC = () => {
+  const [usedKeyboardKey, setUsedKeyboardKey] = useState('');
 
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+  const detectKeyDown = (event: KeyboardEvent) => {
+    setUsedKeyboardKey(event.key);
+  };
+
+  useEffect(() => {
+    document.addEventListener('keydown', detectKeyDown);
+  }, []);
```

Comment:
don't forget to cleanup before component unmounts - a callback after `return` statement
i'd also prefer `keyup` instead of `keydown` - so it doesn't spam events when you hold on the key - but only when press & release it.

```suggestion
  useEffect(() => (
    document.addEventListener('keyup', detectKeyDown)
    
    return () => document.removeEventListener('keyup', detectKeyDown);
  ), []);
```

Reply:
Cleaned it up, and added keyup instead of keydown.